src/document.c

```diff
@@ -1545,6 +1564,21 @@ gboolean document_reload_force(GeanyDocument *doc, const gchar *forced_enc)
 	pos = sci_get_current_position(doc->editor->sci);
 	new_doc = document_open_file_full(doc, NULL, pos, doc->readonly, doc->file_type, forced_enc);
 
+	if (file_prefs.keep_edit_history_on_reload && file_prefs.show_keep_edit_history_on_reload_msg)
+	{
+		bar = document_show_message(doc, GTK_MESSAGE_INFO,
+						on_keep_edit_history_on_reload_response,
+						GTK_STOCK_OK, GTK_RESPONSE_ACCEPT,
+						_("Discard history"), GTK_RESPONSE_NO,
+						NULL, 0, _("The buffer's previous state is stored in the history. "
+						"Undoing restores it. You can disable this by discarding the history upon "
+						"reload. This message will not be displayed again. "
+						"Your choice can be changed in the various preferences."),
```

Comment:
This becomes longer again. It's already unwidely long IMO.

If the number of sentences is the problem why not just join the sentences? "The buffer's previous state is stored in the history and undoing restores it. You can disable this by discarding the history upon reload. This message will not be displayed again but your choice can be changed in the various preferences."